lld/wasm/Relocations.cpp

```diff
@@ -170,6 +170,19 @@ void scanRelocations(InputChunk *chunk) {
         if (requiresGOTAccess(sym))
           addGOTEntry(sym);
         break;
+      case R_WASM_TABLE_INDEX_REL_SLEB:
+      case R_WASM_TABLE_INDEX_REL_SLEB64:
+      case R_WASM_MEMORY_ADDR_REL_SLEB:
+      case R_WASM_MEMORY_ADDR_REL_SLEB64:
+        // These relocation types are only present in the code section and
+        // are not supported. It would require replacing the constant by using
+        // a GOT global.
+        if (sym->isUndefined())
+          error(toString(file) + ": relocation " +
+                relocTypeToString(reloc.Type) +
+                " is not supported against an undefined symbol `" +
+                toString(*sym) + "`");
+        break;
```

Comment:
The above condition looks wrong for this block.   Shouldn't this be just guarded by `sym->isUndefined()` and nothing else?

We don't want to be reporting here is here for defined symbols in PIC mode (for example).    In fact, IIUC, these `REL` relocations are specifically designed for PIC mode.

Reply:
Thank you. I adjusted this.